### PR TITLE
[JSS] Export the JSS class declaration

### PR DESCRIPTION
**Pull request by milesj**
to: @appsforartists @frenic @pelotom 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

It's not possible to use the `JSS` type as an annotation, which would be super useful for something like:

```
constructor(jss: JSS) {
  this.jss = jss;
}
```

Right now I'm doing `type JSS = ReturnType<typeof create>;` just to infer the class.

### CODE REVIEW

File: types/jss/jss-tests.ts [hunk 1]
```diff
@@ -1,18 +1,20 @@
 // API docs at http://cssinjs.org/js-api
```
Comment (milesj) [inline types/jss/jss-tests.ts]: These tests are unrelated but were failing on 2.9.

**Comment by appsforartists** [comment]
Is it possible to split this into separate PRs?  I'm happy to approve the export right away.  Figuring out whether csstype.hyphen is the correct generic is more thinking than I have time for right now. 😃

**Comment by milesj** [comment]
@appsforartists Went with hyphen because of this comment: https://github.com/milesj/DefinitelyTyped/blob/5d2acc01ab583dca20999ecc9d9d04e899066f98/types/jss/jss-tests.ts#L22

I'll undo the test changes for the time being.

File: types/jss/index.d.ts [hunk 2]
```diff
@@ -104,13 +104,13 @@ export interface RuleOptions {
...
-    reset(): void;
-    remove(sheet: StyleSheet): void;
-    toString(options?: ToCssOptions): string;
+	constructor();
+	registry: ReadonlyArray<StyleSheet>;
+	readonly index: number;
+	add(sheet: StyleSheet): void;
+	reset(): void;
+	remove(sheet: StyleSheet): void;
+	toString(options?: ToCssOptions): string;
```
Comment (appsforartists) [inline types/jss/index.d.ts]: Any reason for the style change here?  I think the style on here was dictated by a linter.  In any case, it probably shouldn't change without reason (and 2 or 4 spaces is more common than 3).

**Review by appsforartists — APPROVED** [review]

**Comment by appsforartists** [comment]
Thanks.

File: types/jss/index.d.ts (hunk 2 above)
Comment (milesj) [reply types/jss/index.d.ts]: The rest of the file was tabs, while this part was spaces, so my editor re-formatted it automatically.

**Pull request merged**